metaflow/includefile.py

```diff
@@ -80,6 +83,93 @@ def path(self):
         return self._path
 
 
+class Azure(object):
```

Comment:
nit: maybe this whole this can move over to the azure folder to keep this a little cleaner (kind of like S3 doesn't show up here except in the dictionary?

Reply:
The intent is that this should stay logically private for includefile usage ONLY.  I think we can do something like `metaflow/plugins/azure/includefile_support.py` which would be sufficiently explicit about it.